pretty_printers_tests/tests/ffi_strings.rs

```diff
@@ -1,69 +1,122 @@
-// lldb-command:print empty_os_str
-// lldbr-check:[...]empty_os_str = [...] ""
-// TODO: investigate why `lldbg-check` fails
-// lldb-command:print path1
-// lldbr-check:[...]path1 = [...] "/a/b/∂"
-// TODO: investigate why `lldbg-check` fails
-// lldb-command:print empty_path
-// lldbr-check:[...]empty_path = [...] ""
-// TODO: investigate why `lldbg-check` fails
-// lldb-command:print c_str1
-// lldbr-check:[...]c_str1 = [...] "abcd"
-// TODO: investigate why `lldbg-check` fails
+// lldbr-check:[...]path_buf = "/a/b/c" [...]
+// lldbg-check:[...]$11 = "/a/b/c" [...]
+
+// lldb-command:print path_buf_empty
+// lldbr-check:[...]path_buf_empty = "" [...]
+// lldbg-check:[...]$12 = "" [...]
+
+// lldb-command:print path_buf_unicode
+// lldbr-check:[...]path_buf_unicode = "/a/b/∂" [...]
+// TODO: support WTF-8 on Windows
+
+
+// lldb-command:print path
+// lldbr-check:[...]path = "/a/b/c" { data_ptr = [...] length = 6 }
+// TODO: support `ref$<...>` and `ref_mut$<...>` type wrappings on Windows
+
+// lldb-command:print path_empty
+// lldbr-check:[...]path_empty = "" { data_ptr = [...] length = 0 }
+// TODO: support `ref$<...>` and `ref_mut$<...>` type wrappings on Windows
+
+// lldb-command:print path_unicode
+// lldbr-check:[...]path_unicode = "/a/b/∂" { data_ptr = [...] length = 8 }
+// TODO: support `ref$<...>` and `ref_mut$<...>` type wrappings on Windows
+
 
 // === GDB TESTS ==================================================================================
 
 // gdb-command:run
 
 // gdb-command:print os_string
-// gdb-check:[...]$1 = "A∆й中"
-// gdb-command:print empty_os_string
+// gdb-check:[...]$1 = "abc"
+// gdb-command:print os_string_empty
 // gdb-check:[...]$2 = ""
+// gdb-command:print os_string_unicode
+// gdb-check:[...]$3 = "A∆й中"
 
 
 use std::ffi::{CStr, CString, OsStr, OsString};
 use std::ops::DerefMut;
 use std::path::{Path, PathBuf};
 
 fn main() {
-    let mut os_string = OsString::from("A∆й中");
-    let empty_os_string = OsString::from("");
-    let c_string1 = CString::new("A∆й中").unwrap();
-    let path_buf = PathBuf::from("/a/b/∂");
-    let empty_path_buf = PathBuf::from("");
-    let os_str1 = OsStr::new("A∆й中");
-    let os_str2 = os_string.deref_mut();
-    let empty_os_str = OsStr::new("");
-    let path1 = Path::new("/a/b/∂");
-    let empty_path = Path::new("");
-    let c_str1 = CStr::from_bytes_with_nul(b"abcd\0").unwrap();
+    let mut os_string = OsString::from("abc");
+    let os_string_empty = OsString::from("");
+    let mut os_string_unicode = OsString::from("A∆й中");
+
+    let os_str = os_string.deref_mut();
+    let os_str_empty = OsStr::new("");
+    let os_str_unicode = os_string_unicode.deref_mut();
+
+    let c_string = CString::new("abc").unwrap();
+    let c_string_empty = CString::new("").unwrap();
+    let c_string_unicode = CString::new("A∆й中").unwrap();
+
+    let c_str = CStr::from_bytes_with_nul(b"abcd\0").unwrap();
+    let c_str_empty = CStr::from_bytes_with_nul(b"\0").unwrap();
```

Comment:
Let's also add `let c_str_unicode = CStr::from_bytes_with_nul(b"A\xE2\x88\x86\xD0\xB9\xE4\xB8\xAD\0").unwrap();` line to check unicode for `CStr` and check pretty printer in this case as well